packages/engine/src/datastore/table/state/create_remove/plan.rs

```diff
@@ -68,7 +65,8 @@ impl<'a> MigrationPlan<'a> {
         for (batch_index, action) in self.existing_mutations.iter().enumerate().rev() {
             if let ExistingGroupBufferActions::Remove = action {
                 // Removing in tandem to keep similarly sized batches together
-                removed_ids.push(state_agent_pool.swap_remove(batch_index));
+                removed_ids.push(state.agent_pool.swap_remove(batch_index));
```

Comment:
I don't immediately recall where we use the removed agent batch ids, but do we not also need removed message batch ids there?

Reply:
They are not used at all. I removed `removed_ids`.

Reply:
On second thought, I think it would actually be useful to send removed ids to runners so they can free any loaded data, and it might also be necessary to unmap shared memory manually in the Rust/Python processes, so I would keep both agent and message batch ids and add a TODO about freeing data/unmapping memory.